### DISCUSSION TITLE
Traces for dead jvm

### DISCUSSION THREAD

**Discussion by padington** [async-profiler/async-profiler#431] [Q&A] [answered]
If there any option to get traces from profiling session that have not been finished due to killed jvm?
When I try to stop the session the java process does not exist anymore.
I've seen some issues related to continuous profiling and the 'trace' branch. But it is quite outdated and I'm not sure if it can help.
Any option that I've missed?

Thank you in advance

**Comment by apangin** [comment, marked as answer]
Profiling data is aggregated in the JVM process. When the JVM stops abnormally, the data is lost.

One possible approach is to run short profiling sessions periodically. Profiles in .collapsed and .jfr formats can be easily merged just by concatenating the files. In the worst case you will loose only the last short session.

It's also possible to dump profiling data without ending a profiling session. After dumping the profile, run `resume` command, and the profiler will proceed, retaining all previously collected samples.

I'm going to implement chunked JFR output in the nearest future (#360). This will be a kind of automation for the above approach:  async-profiler will periodically dump JFR chunks without interrupting the session. If the JVM crashes, it will be possible to build the profile from all chunks except the last one.

**Reply by padington** [reply]
Thanks, Andrei, for the extensive answer. Will try periodical sessions.
